### PR TITLE
New diagrams

### PR DESCRIPTION
**Pull request by ralphbean**
I added a new application context diagram and deleted the old one. The old one was out of date and just didn't seem helpful to me.

Also, added some text to the book index to provide some context for each of the service links.

Doing this for https://issues.redhat.com/browse/RHTAPSRE-69

### CODE REVIEW

**Review by adambkaplan — APPROVED** [review]
/approve

One suggestion for Pipeline Service, otherwise lgtm

**Comment by adambkaplan** [comment]
Regarding the "Proxy" rectangle - shouldn't this be two shapes?

- Sprayproxy ("Source Webhook Receiver?")
- The CodeReady toolchain ("Kubernetes Proxy", "Workspace Manager"?)

File: diagrams/appstudio.drawio.svg
Comment (arewm) [inline diagrams/appstudio.drawio.svg:1]: Is there a reason that you are including the same diagram twice with different names?

File: book/index.md [hunk 1]
```diff
@@ -26,33 +26,44 @@ Red Hat developer services provide the platform for building integrated experien
...
+- [Java Rebuilds Service](./jvm-build-service.md) - A subsystem of the build-service that manages
+  the rebuild of binary java jars pulled from maven central for an improved degree of provenance.
+- [Integration Service](./integration-service.md) - A workflow service that manages execution of
+  users' tests and promotion in response to completing builds.
+- [Release Service](./release-service.md) - A workflow service that manages execution of privileged
+  pipelines to release user content to protected destinations.
+- [GitOps Service](./gitops-service.md) - A foundational service providing deployment of user
+  applications.
+- [Pipeline Service](./pipeline-service.md) - A foundational service providing PipelineRuns to other
+  services.
```
[the remark this answers is not in the document]
Comment (adambkaplan) [reply book/index.md:62]: :thinking: I think it makes sense to have a separate document (as a follow up to this PR) that describes at a high level how we implement secure software supply chain. Multiple components contribute to our opinionated implementation:

1. Tekton Chains
2. SBOMs generated in build pipelines
3. Enterprise Contract verification
4. Release Service gating

[the remark this answers is not in the document]
Comment (arewm) [reply book/index.md:62]: ... and how pipeline service fits into it all. :)

File: diagrams/appstudio.drawio.svg
Comment (ralphbean) [reply diagrams/appstudio.drawio.svg:1]: They are different exports from drawio, which I actually don't like, but the draw.io is the tool Gorkem decided on for diagrams in the book (maybe we revise this some day?). I'll delete the non-drawio one for now.

Comment (ralphbean) [reply diagrams/appstudio.drawio.svg:1]: Resolved in d5a8359.

**Review by arewm — APPROVED** [review]

**Comment by cmoulliard** [comment]
If possible, I would like to suggest some improvements:
- Colorize the boxes to better separate visually the CRD (= API) from Service
- Add dashed vertical lines to show which CRD are managed by which services 
- Shade or use a more light color to separate the CRDs that a user can manage (=CRUD) from unmanageable 
- Add link under section "Service (Component) Context" about "CodeReady toolchain"
- The Enterprise Contract Service is missing. Is it a mistake ?

@ralphbean

**Pull request merged**